core/src/main/java/cucumber/runtime/arquillian/ArquillianCucumber.java

```diff
@@ -350,6 +242,73 @@ public void runAfterHooks(final Reporter reporter, final Set<Tag> tags) {
         }
 
         return filters;
+    } 
+    
+    private static List<CucumberFeature> buildFeatureList(final Set<Object> testFilters, final InputStream featuresInputStream, final ClassLoader classLoader, final Set<Entry<String, Collection<URL>>> featuresSet) throws Exception
+    {
+    	final List<CucumberFeature> cucumberFeatures = new ArrayList<CucumberFeature>();
+        final FeatureBuilder featureBuilder = new FeatureBuilder(cucumberFeatures);
+        
+        if (featuresInputStream != null) {
+            final BufferedReader featuresFileReader = new BufferedReader(new InputStreamReader(featuresInputStream));
```

Comment:
maybe I'm missing it - just reading on github the patch - but where is this reader closed? BTW I'd move it in next method since it is not reused then